### PR TITLE
improve tests

### PR DESCRIPTION
**Pull request by dsavitskiy**

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

File: ci.nix [hunk 1]
```diff
@@ -64,6 +64,7 @@ mkShell {
   PROTOC_INCLUDE = io-engine.PROTOC_INCLUDE;
   SPDK_PATH = if nospdk then null else "${libspdk-dev}";
   ETCD_BIN = "${etcd}/bin/etcd";
+  FIO_SPDK = "${libspdk-dev}/fio/spdk_nvme";
```
Comment (tiagolobocastro) [inline ci.nix]: I wonder if we need to check `if no spdk then null` like SPDK_PATH, in case we're using custom spdk?

**Review by tiagolobocastro — APPROVED** [review]

File: ci.nix (hunk 1 above)
Comment (dsavitskiy) [reply ci.nix]: ah, yes. Will fix.

File: io-engine-tests/src/file_io.rs [hunk 2]
```diff
@@ -93,3 +126,65 @@ pub async fn compute_file_checksum(
...
+            break;
+        }
+
+        for i in 0 .. na {
+            if buf_a[i] != buf_b[i] {
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    format!(
+                        "Miscompare at {} ({} blk): {:?} {:#02x} != {:?} {:#02x}",
+                        pos, pos / 512, name_a, buf_a[i], name_b, buf_b[i],
```
Comment (dsharma-dc) [inline io-engine-tests/src/file_io.rs]: This is assuming 512-bytes block size always? fio has configurable block size though.

File: io-engine-tests/src/fio.rs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+        let counter = JOB_COUNTER.fetch_add(1, Ordering::SeqCst);
+
+        Self {
+            counter,
+            name: format!("fio-{counter}"),
+            ioengine: "spdk".to_string(),
+            filename: String::new(),
+            direct: true,
+            blocksize: 4096,
```
Comment (dsharma-dc) [inline io-engine-tests/src/fio.rs]: Here it's hardcoded 4K

**Review by dsharma-dc — APPROVED** [review]

File: io-engine-tests/src/file_io.rs (hunk 2 above)
Comment (dsavitskiy) [reply io-engine-tests/src/file_io.rs]: I'll remove it, it's confusing indeed.

File: io-engine-tests/src/fio.rs (hunk 3 above)
Comment (dsavitskiy) [reply io-engine-tests/src/fio.rs]: It's the same default as fio's. I'll make it optional to avoid confusion.

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**